newIDE/app/src/EventsSheet/EventsTree/Renderers/LinkEvent/ExternalEventsAutoComplete.js

```diff
@@ -8,21 +8,31 @@ import SemiControlledAutoComplete, {
   type DataSource,
 } from '../../../../UI/SemiControlledAutoComplete';
 
-const getList = (project: ?gdProject): DataSource => {
-  if (!project) {
+const getList = (
+  currentSceneName: ?string,
+  currentExternalEventName: ?string,
+  project: ?gdProject
+): DataSource => {
+  if (!project || !currentSceneName) {
     return [];
   }
 
-  const externalEvents = enumerateExternalEvents(project).map(
-    externalEvents => ({
+  const externalEvents = enumerateExternalEvents(project)
+    .filter(
+      externalEvents => externalEvents.getName() !== currentExternalEventName
+    )
+    .map(externalEvents => ({
       text: externalEvents.getName(),
       value: externalEvents.getName(),
-    })
-  );
-  const layouts = enumerateLayouts(project).map(layout => ({
-    text: layout.getName(),
-    value: layout.getName(),
-  }));
+    }));
+
+  const layouts = enumerateLayouts(project)
+    .filter(layout => layout.getName() !== currentSceneName || externalEvents)
```

Comment:
Why this `|| externalEvents`? I don't think this is doing what you think?

Reply:
It's to allow including the events in the scene linked to the externals events like externalevents1 is linked to scene1 but it indeed causes a bug that allow the same never ending loop in the compiler. I am removing it.

Reply:
externalEvents is an array, so it will always be true, so this filter will filter nothing => so whatever you tried to do, it is probably not what you intended. Be sure to double check every line and ensure JavaScript is properly doing what you think ;)